tests/subsys/s11n/fcb/src/s11n_test_fcb.c

```diff
@@ -0,0 +1,350 @@
+	}
+}
+
+void
+config_test_fill_area(char test_value[S11N_TEST_FCB_VAL_STR_CNT]
+				     [S11N_MAX_VAL_LEN],
+		      int iteration)
+{
+	int i, j;
+
+	for (j = 0; j < 64; j++) {
+		for (i = 0; i < S11N_MAX_VAL_LEN; i++) {
+			test_value[j][i] = ((j * 2) + i + iteration) % 10 + '0';
+		}
+		test_value[j][sizeof(test_value[j]) - 1] = '\0';
+	}
+}
+
+char *c2_var_find(char *name)
+{
+	int idx = 0;
+	int len;
+	char *eptr;
+
+	len = strlen(name);
+	zassert_true(len > 6, "string type expected\n");
+	zassert_true(!strncmp(name, "string", 6), "string type expected\n");
+
+	idx = strtoul(&name[6], &eptr, 10);
+	zassert_true(*eptr == '\0', "EOF\n");
+	zassert_true(idx < c2_var_count,
+		     "var index greather than any exporter\n");
+
+	return val_string[idx];
+}
+
+char *c2_handle_get(int argc, char **argv, char *val, int val_len_max)
+{
+	int len;
+	char *valptr;
+
+	if (argc == 1) {
+		valptr = c2_var_find(argv[0]);
+		if (!valptr) {
+			return NULL;
+		}
+
+		len = strlen(val_string[0]);
+		if (len > val_len_max) {
+			len = val_len_max;
+		}
+
+		strncpy(val, valptr, len);
+	}
+
+	return NULL;
+}
+
+int c2_handle_set(int argc, char **argv, char *val)
+{
+	char *valptr;
+
+	if (argc == 1) {
+		valptr = c2_var_find(argv[0]);
+		if (!valptr) {
+			return -ENOENT;
+		}
+
+		if (val) {
+			strncpy(valptr, val, sizeof(val_string[0]));
+		} else {
+			memset(valptr, 0, sizeof(val_string[0]));
+		}
+
+		return 0;
+	}
+
+	return -ENOENT;
+}
+
+int c2_handle_export(void (*cb)(char *name, char *value),
+		     enum s11n_export_tgt tgt)
+{
+	int i;
+	char name[32];
+
+	for (i = 0; i < c2_var_count; i++) {
+		snprintf(name, sizeof(name), "2nd/string%d", i);
+		cb(name, val_string[i]);
+	}
+
+	return 0;
+}
+
+char *c3_handle_get(int argc, char **argv, char *val, int val_len_max)
+{
+	if (argc == 1 && !strcmp(argv[0], "v")) {
+		return s11n_str_from_value(S11N_INT32, &val32, val,
+					   val_len_max);
+	}
+
+	return NULL;
+}
+
+int c3_handle_set(int argc, char **argv, char *val)
+{
+	u32_t newval;
+	int rc;
+
+	if (argc == 1 && !strcmp(argv[0], "v")) {
+		rc = S11N_VALUE_SET(val, S11N_INT32, newval);
+		zassert_true(rc == 0, "S11N_VALUE_SET callback");
+		val32 = newval;
+		return 0;
+	}
+
+	return -ENOENT;
+}
+
+int c3_handle_export(void (*cb)(char *name, char *value),
+		     enum s11n_export_tgt tgt)
+{
+	char value[32];
+
+	s11n_str_from_value(S11N_INT32, &val32, value, sizeof(value));
+	cb("3/v", value);
+
+	return 0;
+}
+
+void config_empty_lookups(void);
+void config_test_insert(void);
+void config_test_getset_unknown(void);
+void config_test_getset_int(void);
+void config_test_getset_bytes(void);
+void config_test_getset_int64(void);
+void config_test_commit(void);
+
+void config_test_empty_fcb(void);
+void config_test_save_1_fcb(void);
+void config_test_insert2(void);
+void config_test_save_2_fcb(void);
+void config_test_insert3(void);
+void config_test_save_3_fcb(void);
+void config_test_compress_reset(void);
+void config_test_save_one_fcb(void);
```

Comment:
can you please use test_config_... naming convention instead? This is how we have been naming this almost everywhere and trying now to create some consistency to easy parsing when generating reports.